src/cursor/cur_index.c

```diff
@@ -347,6 +354,45 @@ __curindex_search_near(WT_CURSOR *cursor, int *exact)
     API_END_RET(session, ret);
 }
 
+/*
+ * __increment_bound_array --
+ *     Increment the given buffer by one bit, and set the increment variable to true. If the all of
+ *     the values inside the buffer is are UINT8_MAX, we do not increment the buffer, and we return
+ *     setting the increment variable to false.
+ */
+static inline int
+__increment_bound_array(WT_ITEM *user_item, bool *increment)
+{
+    size_t usz;
```

Comment:
Mixing size_t and int like this is a bit concerning, I understand that you want the int to be able to be negative but with a theoretically big enough usz you'd run into trouble when you cast it to an int. 

I think we may as well define usz as int at the start of the function and do the cast once. Then give a reason as to why its okay?